cmd/src/server.rs

```diff
@@ -1,442 +1,548 @@
-    let (mut worker, resolver) = resolve::new_resolver(Arc::clone(&pd_client))
-        .unwrap_or_else(|e| fatal!("failed to start address resolver: {}", e));
-    let pd_sender = pd_worker.scheduler();
-
-    // Create encrypted env from cipher file
-    let encrypted_env = if !cfg.security.cipher_file.is_empty() {
-        match encrypted_env_from_cipher_file(&cfg.security.cipher_file, None) {
-            Err(e) => fatal!(
-                "failed to create encrypted env from cipher file, err {:?}",
-                e
-            ),
-            Ok(env) => Some(env),
-        }
-    } else {
-        None
-    };
+struct Engines {
+    engines: engine::Engines,
+    store_meta: Arc<Mutex<StoreMeta>>,
+    engine: RaftKv<ServerRaftStoreRouter>,
+    raft_router: ServerRaftStoreRouter,
+}
 
-    // Create block cache.
-    let cache = cfg.storage.block_cache.build_shared_cache();
+struct Servers {
+    lock_mgr: Option<LockManager>,
+    region_info_accessor: RegionInfoAccessor,
+    server: Server<ServerRaftStoreRouter, resolve::PdStoreAddrResolver>,
+    node: Node<RpcClient>,
+    importer: Arc<SSTImporter>,
+}
 
-    // Create raft engine.
-    let mut raft_db_opts = cfg.raftdb.build_opt();
-    if let Some(ref ec) = encrypted_env {
-        raft_db_opts.set_env(ec.clone());
-    }
-    let raft_db_cf_opts = cfg.raftdb.build_cf_opts(&cache);
-    let raft_engine = rocks::util::new_engine_opt(
-        raft_db_path.to_str().unwrap(),
-        raft_db_opts,
-        raft_db_cf_opts,
-    )
-    .unwrap_or_else(|s| fatal!("failed to create raft engine: {}", s));
-
-    // Create kv engine, storage.
-    let mut kv_db_opts = cfg.rocksdb.build_opt();
-    kv_db_opts.add_event_listener(compaction_listener);
-    if let Some(ec) = encrypted_env {
-        kv_db_opts.set_env(ec);
+impl TiKVServer {
+    fn init(mut config: TiKvConfig) -> TiKVServer {
+        let store_path = Path::new(&config.storage.data_dir).to_owned();
+
+        let security_mgr =
+            Arc::new(SecurityManager::new(&config.security).unwrap_or_else(|e| {
+                fatal!("failed to create security manager: {}", e.description())
+            }));
+
+        let pd_client = Self::connect_to_pd_cluster(&mut config, security_mgr.clone());
+
+        let (resolve_worker, resolver) = resolve::new_resolver(Arc::clone(&pd_client))
+            .unwrap_or_else(|e| fatal!("failed to start address resolver: {}", e));
+
+        // Initialize raftstore channels.
+        let (router, system) = fsm::create_raft_batch_system(&config.raft_store);
+
+        TiKVServer {
+            config,
+            security_mgr,
+            pd_client,
+            router,
+            system: Some(system),
+            resolver,
+            store_path,
+            engines: None,
+            servers: None,
+            to_stop: vec![Box::new(resolve_worker)],
+        }
     }
 
-    // Before create kv engine we need to check whether it needs to upgrade from v2.x to v3.x.
```

Comment:
Better keep the comment. It's expected to be uncommented in the future.

Reply:
Can we make this an issue instead? Leaving commented-out code checked-in is messy, and there is no context here as to when or if it should be re-enabled or if it is a mistake.

Reply:
I think the purpose is not to implement the logic again.

/cc @overvenus PTAL

Reply:
Please remove it, there is no short term plan for the upgrading.